api/v1alpha1/addresspool_types.go

```diff
@@ -40,6 +53,11 @@ type AddressPoolSpec struct {
 	// +optional
 	// +kubebuilder:default:=true
 	AutoAssign *bool `json:"autoAssign,omitempty" yaml:"auto-assign,omitempty"`
+
+	// When an IP is allocated from this pool, how should it be
+	// translated into BGP announcements?
+	// +optional
+	BGPAdvertisements []BgpAdvertisement `json:"bgpAdvertisements,omitempty" yaml:"bgp-advertisements,omitempty"`
```

Comment:
Does it make sense for there to ever be more than one `BgpAdvertisement`?  I know the ConfigMap has it represented this way, but I'm not sure it's necessary.

Reply:
I am not sure either @russellb so far most of the configs I have seen its been using one instance , I made it slice to align with Metallb and added UT to check multiple instance but if you feel strongly it will never be more than one I can change it

Reply:
I just read the MetalLB code and it *does* allow and make use of multiple advertisement configs.  It will do multiple advertisements for the same IP using the different configurations if you configure it that way.  I guess we should leave it how you have it!